apis/v1alpha2/validation/httproute_test.go

```diff
@@ -431,3 +431,119 @@ func TestValidateHTTPPathMatch(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateServicePort(t *testing.T) {
+	portPtr := func(n int) *gatewayv1a2.PortNumber {
+		p := gatewayv1a2.PortNumber(n)
+		return &p
+	}
+
+	groupPtr := func(g string) *gatewayv1a2.Group {
+		p := gatewayv1a2.Group(g)
+		return &p
+	}
+
+	kindPtr := func(k string) *gatewayv1a2.Kind {
+		p := gatewayv1a2.Kind(k)
+		return &p
+	}
+
+	tests := []struct {
+		name     string
+		route    *gatewayv1a2.HTTPRoute
+		errCount int
+	}{
+		{
+			name:     "default groupkind with port",
+			errCount: 0,
+			route: &gatewayv1a2.HTTPRoute{
+				Spec: gatewayv1a2.HTTPRouteSpec{
+					Rules: []gatewayv1a2.HTTPRouteRule{{
+						BackendRefs: []gatewayv1a2.HTTPBackendRef{{
+							BackendRef: gatewayv1a2.BackendRef{
+								BackendObjectReference: gatewayv1a2.BackendObjectReference{
+									Port: portPtr(99),
```

Comment:
Isn't this an invalid ObjectReference anyway because it is missing the required Name and Kind (Kind will defaulted _usually_)? I guess it is okay to unit test it.

Reply:
Yeh, there's a case for that. I felt bad about assuming the fields were non-nil in the validator, since there was an argument that we can't guarantee that the API server has validated things. And then I wanted to test that case.